cycles-ledger/src/storage.rs

```diff
@@ -1184,16 +1184,36 @@ mod create_canister {
 
     use super::transfer_from::UNKNOWN_GENERIC_ERROR;
 
-    pub fn anyhow_error(error: anyhow::Error) -> CreateCanisterError {
+    pub fn unknown_generic_error(message: String) -> CreateCanisterError {
+        CreateCanisterError::GenericError {
+            error_code: Nat::from(UNKNOWN_GENERIC_ERROR),
+            message,
+        }
+    }
+}
+
+mod create_canister_from {
+    use super::transfer_from::{CANNOT_TRANSFER_FROM_ZERO, UNKNOWN_GENERIC_ERROR};
+    use crate::endpoints::CreateCanisterFromError;
+    use candid::Nat;
+
+    pub fn anyhow_error(error: anyhow::Error) -> CreateCanisterFromError {
         unknown_generic_error(format!("{:#}", error))
     }
 
-    pub fn unknown_generic_error(message: String) -> CreateCanisterError {
-        CreateCanisterError::GenericError {
+    pub fn unknown_generic_error(message: String) -> CreateCanisterFromError {
+        CreateCanisterFromError::GenericError {
             error_code: Nat::from(UNKNOWN_GENERIC_ERROR),
             message,
         }
     }
+
+    pub fn cannot_create_with_zero() -> CreateCanisterFromError {
```

Comment:
Isn't this simply a special case of insufficient funds?
Why should this case be handled separately?

Reply:
No, this is about creating a canister using 0 cycles no matter what the account balance is